### DISCUSSION TITLE
Minimalizing the codebase

### DISCUSSION THREAD

**Discussion by nordkorea** [searxng/searxng#2411] [Q&A]
I'd like to make the codebase as simple as possible by removing "unnecessary" files that do not break core functionality if it does not exist, I am planning to host it with some modifications on Hostinger or another hosting service provider. Here are the ones I have deleted:

I wanted to ask if there are more of them that I can delete? Or if the ones I have deleted will break the core functionality.
Any assistance is appreciated since I am kind of obsessed with minimalism. Bloat-free codebase but functional and stable is all I ask for.

DELETED FILES:
./docs
./examples
./.devcontainer
./.github
./.vscode
./AUTHORS.rst
./CHANGELOG.rst
./CONTRIBUTING.md
./LICENSE
./PULL_REQUEST_TEMPLATE.md
./README.rst
./SECURITY.md
./.gitattributes
./.gitignore
./.weblate
./searx/engines/docker_hub.py
./dockerfiles
Dockerfile
.dockerignore
./searxng_extra
./searx/templates
searx/translations

+ plus

deleted all files incl. the name "makefile"
(13 of them in total)

**Comment by allendema** [comment]
The .md and .git* files contain just text, so a few hundred kb.

Leave the ```searx/templates``` and ```searx/engines``` as is.

Then you can check if something breaks for day to day searching.
